### ISSUE TITLE
Failed to run `config` for snipe.nvim

### ISSUE THREAD

**Issue by lisez** [leath-dub/snipe.nvim#47]
I faces this error after updated to latest version: [be62bce](https://github.com/leath-dub/snipe.nvim/commit/be62bce1a0069435de0cbcd6ebe9b44054da19db)

Full error messages:
```bash
Failed to run `config` for snipe.nvim

...(My home dir)/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:112: opt: expected table, got string

# stacktrace:
  - vim/shared.lua:0 _in_ **validate**
  - /snipe.nvim/lua/snipe/init.lua:112 _in_ **setup_config**
  - /snipe.nvim/lua/snipe/init.lua:5 _in_ **setup**
  - ~/.config/nvim/init.lua:1318
```

I use lazy.nvim:
```lua
	{
		"leath-dub/snipe.nvim",
		opts = {
			ui = {
				position = "bottomright",
			},
			navigate = {
				cancel_snipe = "q",
			},
			sort = "last",
		},
	},
```

nvim version info:
```
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1732813678
```

**Comment by iDefyU** [comment]
the same issue occurred after the update

```
E5108: Error executing lua: ...ker/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:259: attempt to index field 'config' (a nil value)
stack traceback:
	...ker/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:259: in function 'open_buffer_menu'
	/Users/ryker/.config/nvim/lua/plugins/snipe.lua:7: in function </Users/ryker/.config/nvim/lua/plugins/snipe.lua:6>
```

```
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1732813678
```

**Comment by shouyierban** [comment]
same  issue

**Comment by kungfusheep** [comment]
Hey - Looks to be this new call to `validate`. If I comment this block out it works fine. 

https://github.com/leath-dub/snipe.nvim/blame/be62bce1a0069435de0cbcd6ebe9b44054da19db/lua/snipe/init.lua#L112

<img width="923" alt="image" src="https://github.com/user-attachments/assets/57288ded-b03b-4e1a-b3e3-d330999ef0e3">

**Comment by GusJelly** [comment]
#48 Fixes it for me as well, the validation also continues to find errors correctly with the change.

**Comment by leath-dub** [comment]
OMG! Im so sorry, I don't know why I cant see that issue on my configuration. I changed that vim.validate to fix deprecation warnings.

**Comment by leath-dub** [comment]
Ohh I see! sorry it is because I am on neovim nightly, the API changed. Very sorry about that I will be more careful about API changes in the future.

**Comment by leath-dub** [comment]
I just merged #48 thanks @nicholasxjy for the quick fix. Basically `vim.validate` is being deprecated in the way I was using it before, but the change is only in nightly currently.
